target/linux/ramips/dts/ApexeraNetBot.dts

```diff
@@ -0,0 +1,183 @@
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "apexera-netbot", "ralink,mt7620a-soc";
+	model = "Apexera NetBot";
+
+	chosen {
+		bootargs = "console=ttyS1,57600";
+        };
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		wifi {
+			label = "netbot:led:wifi";
+			gpios = <&gpio2 27 1>;
+		};
+
+		wan {
+			label = "netbot:led:wan";
+			gpios = <&gpio2 26 1>;
+		};
+
+		usb2g {
+			label = "netbot:green:usb2";
+			gpios = <&gpio2 1 1>;
+		};
+
+		usb2r {
+			label = "netbot:red:usb2";
+			gpios = <&gpio2 0 1>;
+		};
+
+		usb4g {
+			label = "netbot:green:usb4";
+			gpios = <&gpio2 3 1>;
+		};
+
+		usb4r {
+			label = "netbot:red:usb4";
+			gpios = <&gpio2 2 1>;
+		};
+
+		usb1g {
+			label = "netbot:green:usb1";
+			gpios = <&gpio2 21 1>;
+		};
+
+		usb1r {
+			label = "netbot:red:usb1";
+			gpios = <&gpio2 22 1>;
+		};
+
+		usb3g {
+			label = "netbot:green:usb3";
+			gpios = <&gpio0 22 1>;
+		};
+
+		usb3r {
+			label = "netbot:red:usb3";
+			gpios = <&gpio0 23 1>;
+		};
+	};
+};
+
+&i2s {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&i2c {
+	status = "okay";
+
+	pcf8563: rtc@51 {
+		status = "okay";
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
+
+	wm8978: wm8978@1A {
+		compatible = "wm8978";
+		reg = <0x1A>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "w25q128";
+		reg = <0 0>;
+		linux,modalias = "m25p80", "w25q128";
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+	    	partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+		        };
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
+		};
+	};
+};
+
+&sdhci {
+	status = "okay";
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ohci {
+	status = "okay";
+};
+
+&pcie {
+	status = "okay";
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x10>;
+	ralink,port-map = "wllll";
+};
+
+&wmac {
+	ralink,mtd-eeprom = <&factory 0>;
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "rgmii1", "rgmii2","mdio","ephy";
+			ralink,function = "gpio";
+		};
+
+		i2s_uartf {
+        ralink,group = "uartf";
```

Comment:
please fix the indention. Please check the whole file for spaces vs. tabs isues.